### ISSUE TITLE
Request: new option to integrate srt into file

### ISSUE THREAD

**Issue by tonystower** [Ascoware/get-iplayer-automator#192]
I currently use subler to add the srt file to the mp4. It would be much easier to have this automated as part of GiA.

I think it is already an option in get_iplayer

Thanks for your consideration.

**Comment by hugofalkman** [comment]
I second that request

**Comment by Vangelis66** [comment]
> I think it is already an option in get_iplayer

... You are right! 👍 
```
--subs-embed    Embed soft subtitles in MP4 output file. Ignored with --audio-only and --ffmpeg-obsolete.
```
It was implemented in 3.13-dev; see 
https://github.com/get-iplayer/get_iplayer/commit/c0d7521
https://github.com/get-iplayer/get_iplayer/commit/c531691

**Comment by raveem** [comment]
Urgh. Created a new thread on this by accident, now closed/deleted once I discovered this one. Here's what I wrote, echoing the above:

_Given that m4v/mp4 containers natively support soft subtitles as part of the standard, there is no reason not to have them embedded in the container. Otherwise, one has to use third party tools to stitch together every single file after download, making the "add to iTunes after download" option pretty useless (iTunes itself, of course, also can use the soft subs capability).

get_iplayer has already implemented this, so this frontend inheriting that switch/capability to download a single file with the subs inside would be a logical step. Please see: https://forums.squarepenguin.co.uk/thread-1606.html_

**Comment by hugofalkman** [comment]
Yes PLEASE,
am spending a lot of useless time with Subler.

**Comment by Vangelis66** [comment]
... Since the feature is already implemented in the **get_iplayer** script itself (see [previous](https://github.com/Ascoware/get-iplayer-automator/issues/192#issuecomment-403668509) comment), in theory _it should be possible_ to add it as a permanent user option to get_iplayer, i.e. in the CLI type: 
`get_iplayer --subs-embed --prefs-add`

However, in the case of GiA on a Mac (i.e. not using the standalone get_iplayer app), one must compose a different incantation to invoke the embedded get_iplayer perl script inside GiA; from previous exchanges, I can recollect something along the following lines (with GiA exited, issue in the Terminal): 

`PERL5LIB="/Applications/Get iPlayer Automator.app/Contents/Resources/perl5" "/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer" --profile-dir="$HOME/Library/Application Support/Get iPlayer Automator" --subs-embed --prefs-add`

To undo that change (delete the permanent option), you'd have to type in the Teminal (again, GiA is exited): 

`PERL5LIB="/Applications/Get iPlayer Automator.app/Contents/Resources/perl5" "/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer" --profile-dir="$HOME/Library/Application Support/Get iPlayer Automator" --subs-embed --prefs-del`

**DISCLAIMER**: _Not tested_ 😉 
(If you're not feeling adventurous, wait for someone more knowledgeable and/or @skovatch to chime in...)

**Comment by skovatch** [comment]
> PERL5LIB="/Applications/Get iPlayer Automator.app/Contents/Resources/perl5" "/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer" --profile-dir="$HOME/Library/Application Support/Get iPlayer Automator" --subs-embed --prefs-add

Yes, this will work. If you also have the command line get_iplayer installed you don't even need the full path to the embedded get_iplayer and PERL5LIB; `--profile-dir` is the important argument.

What do you want this feature to do? The easiest thing to do is to use --subs-embed all the time if subtitles are requested, but I can also add another checkbox for those that just want the raw subtitles.

**Comment by hugofalkman** [comment]
@skovatch Think a message is enough so you know if subtitles are imbedded or if for some reason subtitles wern't provided. (Know this won't apply to ITV, but designing ahead, it's quite common for ITV  not providing subtitles for reruns of older series.)

**Comment by raveem** [comment]
For myself, I would certainly want the equivalent of --subs-embed on all the time (since no functionality is lost if subs come embedded in the file), with an additional option to download the SRT for those that need it.

**Comment by hugofalkman** [comment]
👍
